src/printer.js

```diff
@@ -562,7 +562,20 @@ function genericPrintNoParens(path, options, print) {
       parts.push("return");
 
       if (n.argument) {
-        parts.push(" ", path.call(print, "argument"));
+        if (n.argument.comments) {
+          parts.push(
+            n.loc.start.line === n.loc.end.line
```

Comment:
Not sure to get it, do you mean one giant comment line? I tried to play with it and it looks fine so far.